src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php

```diff
@@ -33,12 +34,14 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
         }
 
         $name = $attribute->name ?? $argument->getName();
+        $validationFailedCode = $attribute->validationFailedStatusCode;
+
         if (!$request->query->has($name)) {
             if ($argument->isNullable() || $argument->hasDefaultValue()) {
                 return [];
             }
 
-            throw new NotFoundHttpException(sprintf('Missing query parameter "%s".', $name));
+            throw new HttpException($validationFailedCode, sprintf('Missing query parameter "%s".', $name));
```

Comment:
It'd be great to preserve the original http exception type.
I've submitted #53209 to help doing so.

Reply:
I think you meant #53212 :)

So should I wait for it to get to merged and base this one on top of 7.1 and consider it as a new feature?

Reply:
it's a new feature yes, please rebase

Reply:
Rebased. I'll also update the code to preserve the original HTTP exception type once #53212 gets merged and I'll mark this one as resolved afterward.

Reply:
Rebase unlocked :)

Reply:
Rebased. I believe we should also fix the labels for the PR.